libs/platform/core/src/services/storage/default-indexed-db-storage.service.ts

```diff
@@ -1,52 +1,73 @@
   indexedDbStorageName,
   IndexedDBStorageService,
   indexedDbTableName,
 } from './indexed-db-storage.service';
-import { StoredValue } from './model';
 
+/** @deprecated since 1.2 */
 export class DefaultIndexedDBStorageService implements IndexedDBStorageService {
-  protected storage = new Observable<Dexie>((subscriber) => {
-    const db = new Dexie(indexedDbStorageName);
-    db.version(1).stores({
-      [indexedDbTableName]: '&key,value',
-    });
-
-    subscriber.next(db);
-
-    return () => {
-      db.close();
-    };
-  }).pipe(
-    switchMap(async (db) => {
-      await db.open();
-
-      return db.table(indexedDbTableName);
-    }),
-    shareReplay({ refCount: false, bufferSize: 1 })
-  );
-
-  getItem(key: string): Observable<StoredValue> {
-    return this.storage.pipe(
-      switchMap((storage) =>
-        liveQuery<StoredValue>(async () => {
-          return (await storage.get(key))?.value;
-        })
+  protected db$: Observable<IDBDatabase> = new Observable(
+    (subscriber: Subscriber<IDBDatabase>) => {
+      const request = indexedDB.open(indexedDbStorageName, 1);
+
+      request.onerror = () => subscriber.error('DB: error');
+      request.onupgradeneeded = (evt: IDBVersionChangeEvent) => {
+        const db = (evt.target as IDBOpenDBRequest).result;
+        db.createObjectStore(indexedDbTableName, { keyPath: 'key' });
+      };
+      request.onsuccess = () => {
+        subscriber.next(request.result);
+        subscriber.complete();
+      };
+    }
+  ).pipe(shareReplay(1));
+
+  protected transactionRequest<T>(
+    mode: IDBTransactionMode,
+    storeAction: (store: IDBObjectStore) => IDBRequest<T>
+  ): Observable<T> {
+    return this.db$.pipe(
+      switchMap(
+        (db) =>
+          new Observable<T>((subscriber) => {
+            const transaction = db.transaction(indexedDbTableName, mode);
+            const objectStore = transaction.objectStore(indexedDbTableName);
+            const request = storeAction(objectStore);
+
+            request.onerror = () =>
+              subscriber.error('IndexedDB: transaction error');
+            request.onsuccess = () => {
+              subscriber.next(request.result);
+              subscriber.complete();
+            };
+          })
       )
     );
   }
 
-  setItem(key: string, value: string): void {
-    this.storage.subscribe((storage) => storage.put({ key, value }));
+  getItem(key: string): Observable<string | null> {
+    return this.transactionRequest('readonly', (store) => store.get(key)).pipe(
+      map((x) => x.value)
+    );
   }
 
-  removeItem(key: string): void {
-    this.storage.subscribe((storage) => storage.delete(key));
+  setItem(key: string, value: string): Observable<unknown> {
+    return subscribeReplay(
+      this.transactionRequest('readwrite', (store) => store.put({ key, value }))
+    );
   }
 
-  clear(): void {
-    this.storage.subscribe((storage) => storage.clear());
+  removeItem(key: string): Observable<unknown> {
```

Comment:
Should we return something when delete or clear?  Prev interfaces were without return values

Reply:
It's deprecated, but I wanted to make it more correct, if someone will use it anyway. I assumed, it's not a breaking change, as customer doesn't expect return anyway.
Also previous implementation was not so correct, as all methods (remove, clear, set) where asynchronous, so it was not possible to wait for them to finish properly.